torch/distributed/checkpoint/optimizer.py

```diff
@@ -49,23 +51,25 @@
 ]
 
 
-def _gen_rank_device(global_rank: int) -> str:
-    if torch.cuda.is_available():
-        return f"cuda:{global_rank % torch.cuda.device_count()}"
+def _gen_rank_device(global_rank: int, device_type: str = "cuda") -> str:
+    device_module = _get_device_module(device_type)
```

Comment:
This is wrong as it will produce `cpu:0` for `cpu`.

We need to add a helper to `torch.cuda` and `torch.cpu` to format a number into a device string.